server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -253,7 +257,8 @@ public EngineConfig(
             retentionLeasesSupplier,
             primaryTermSupplier,
             tombstoneDocSupplier,
-            false
+            false,
+            new InternalTranslogFactory()
```

Comment:
Don't we need similar check that is added for IndexService here as well?
```
this.indexSettings.isRemoteTranslogStoreEnabled() ? null : new InternalTranslogFactory()
```

Reply:
This is the default constructor for EngineConfig